ui/utils/fileDownloader.js

```diff
@@ -12,6 +12,10 @@ export default function downloadFile({ id, type,name, source_type }) {
     dataUri = `/api/pattern/download/${id}`;
   }
 
+  if (type === "filter") {
```

Comment:
As we are having a bunch of these if statements , we might try using a object to map type to path  , also having source_type and type together make the function little complex , hence maybe be prone to bugs ( something to consider)

Reply:
Sounds appropriate to me. @aabidsofi19 will you jump in with a code suggestion (make a commit to this PR/branch)?